libV2/utils/shard.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Returns a unix style timestamp floored to 10 second resolution
+ * @param {Number} timestamp - Unix timestamp with millisecond resolution
+ * @returns {Number} - Unix timestamp representing beginning of shard
+ */
+function shardFromTimestamp(timestamp) {
+    // Do this with string manipulation to avoid float errors
```

Comment:
You can also use Math functions like `Math.floor()` that are meant to deal with int/float conversions. String manipulation works though, just could be slightly cleaner.

Reply:
I've thought of a way to do it with no floats.